src/jsrs_serializer.cc

```diff
@@ -0,0 +1,211 @@
+      value->IsUndefined() ||
+      value->IsNull()) {
+    return value->ToString(isolate->GetCurrentContext()).ToLocalChecked();
+  } else if (value->IsDate()) {
+    return StringifyDate(isolate, value.As<Date>());
+  } else if (value->IsString()) {
+    return StringifyString(isolate, value.As<String>());
+  } else if (value->IsArray()) {
+    return StringifyArray(isolate, value.As<Array>());
+  } else if (value->IsObject()) {
+    return StringifyObject(isolate, value.As<Object>());
+  } else {
+    return Local<String>();
+  }
+}
+
+Local<String> StringifyDate(Isolate* isolate, Local<Date> date) {
+  auto context = isolate->GetCurrentContext();
+  auto toISOString = date->Get(context,
+      String::NewFromUtf8(isolate, "toISOString")).ToLocalChecked();
+  auto result = toISOString.As<Function>()->Call(context,
+      date, 0, nullptr).ToLocalChecked();
+  auto quotes = String::NewFromUtf8(isolate, "\'");
+  auto res_str = result->ToString();
+  res_str = String::Concat(quotes, String::Concat(res_str, quotes));
+  return res_str->ToString();
+}
+
+Local<String> StringifyArray(Isolate* isolate, Local<Array> array) {
+  auto comma = String::NewFromUtf8(isolate, ",");
+  auto result = String::NewFromUtf8(isolate, "[");
+  uint32_t length = array->Length();
+
+  for (uint32_t index = 0; index < length; index++) {
+    auto value = array->Get(index);
+    if (!value->IsUndefined()) {
+      auto chunk = Stringify(isolate, value);
+      if (chunk.IsEmpty()) continue;
+      result = String::Concat(result, chunk);
+    }
+    if (index != length - 1) {
+      result = String::Concat(result, comma);
+    }
+  }
+
+  result = String::Concat(result, String::NewFromUtf8(isolate, "]"));
+  return result;
+}
+
+static const char* GetEscapedControlChar(char str, size_t* size);
+
+Local<String> StringifyString(Isolate* isolate, Local<String> string) {
+  uint32_t length = string->Length();
+
+  std::vector<char> result_str;
+  result_str.reserve((length + 1) * 2);
+  result_str.push_back('\'');
+
+  String::Utf8Value utf8string(string);
+  const char* c_string = *utf8string;
+
+  for (uint32_t i = 0; i < length; i++) {
+    size_t offset;
+    const char* ch = GetEscapedControlChar(c_string[i], &offset);
+    if (ch) {
+      for (size_t k = 0; k < offset; k++) {
+        result_str.push_back(ch[k]);
+      }
+    } else {
+      result_str.push_back(c_string[i]);
+    }
+  }
+
+  result_str.push_back('\'');
+
+  return String::NewFromUtf8(isolate, result_str.data(),
+      NewStringType::kNormal,
+      static_cast<int>(result_str.size())).ToLocalChecked();
+}
+
+// Returns a string representing an escaped control character.
+// If the given character is not a control one, returns nullptr.
+// The `size` is being incremented by the length of the resulting
+// string, but always at least by two, even when the function
+// returns nullptr.
+static const char* GetEscapedControlChar(char str, size_t* size) {
+  static const char* control_chars[0x20] = {
```

Comment:
@belochub and get a `warning: ISO C++11 does not allow conversion from string literal to
      'char *const'` 😉